### PR TITLE
fix: #1056, dedup environment file reading

### PR DESCRIPTION
**Pull request by attilah**
*Issue #, if available:*

#1056

*Description of changes:*

- Add uncaught error handler to CLI for friendlier error messages, due to the lack of GlueGun capability for this
- Throw friendly error message when environment seems to be uninitialized for various commands
- Deduplicate the code when reading the environment file
- Fix some typos

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-cli/src/cli.js [hunk 1]
```diff
@@ -6,6 +6,17 @@ const globalPrefix = require('./lib/global-prefix');
 
 const MIGRATE = 'migrate';
 
+// Gluegun has no hooks for errors happening inside executing the commands,
+// so we need a process level handler, to log the errors like environment
+// not initialized and such errors in a friendly way without technical data
+// like stacktrace which does not add value in this case.
+process.on('uncaughtException', handleError);
+
+function handleError(error) {
```
Comment (kaustavghosh06) [inline packages/amplify-cli/src/cli.js]: Will this show some messages twice?

File: packages/amplify-cli/src/extensions/amplify-helpers/get-env-info.js [hunk 2]
```diff
@@ -7,6 +7,9 @@ function getEnvInfo() {
   let envInfo = {};
   if (fs.existsSync(envFilePath)) {
     envInfo = readJsonFile(envFilePath);
```
Comment (kaustavghosh06) [inline packages/amplify-cli/src/extensions/amplify-helpers/get-env-info.js]: We can use the getEnvInfo() out here right as well right?

File: packages/amplify-cli/src/extensions/amplify-helpers/get-project-details.js [hunk 3]
```diff
@@ -12,11 +13,7 @@ function getProjectDetails() {
     amplifyMeta = readJsonFile(amplifyMetaFilePath);
   }
 
-  let localEnvInfo = {};
-  const envFilePath = pathManager.getLocalEnvFilePath();
-  if (fs.existsSync(envFilePath)) {
-    localEnvInfo = readJsonFile(envFilePath);
-  }
+  const localEnvInfo = getEnvInfo();
```
Comment (kaustavghosh06) [inline packages/amplify-cli/src/extensions/amplify-helpers/get-project-details.js]: What if the file doesn't exist? Will it throw an error right? 
I think in the previous logic we wern't throwing an error when file doesn't exist and just checking for its existence.

File: packages/amplify-cli/src/cli.js (hunk 1 above)
Comment (attilah) [reply packages/amplify-cli/src/cli.js]: It is only for uncaught errors, generally speaking: no, but we do have a few places in our ```catch``` blocks where we ```console.log``` and ```throw``` afterwards, but cannot be determined if there is a higher catch for those blocks or not.

IMHO refactoring the error handling to be true SEH is a different issue, that should be visited in a separate PR for the new platform.

File: packages/amplify-cli/src/extensions/amplify-helpers/get-env-info.js (hunk 2 above)
Comment (attilah) [reply packages/amplify-cli/src/extensions/amplify-helpers/get-env-info.js]: What do you mean, this IS ```getEnvInfo``` itself.

File: packages/amplify-cli/src/extensions/amplify-helpers/get-project-details.js (hunk 3 above)
Comment (attilah) [reply packages/amplify-cli/src/extensions/amplify-helpers/get-project-details.js]: I visited all the places where we were reading this files and we're consuming straight ahead what was returned, without checking it, that's why I added the throw to ```getEnvInfo``` itself. LMK if you find a place where it is not the case, I could missed that, but tried to be thorough.

**Pull request merged**